### ISSUE TITLE
UTF-8 with Chinese can not be rendered

### ISSUE THREAD

**Issue by qinwf** [rstudio/rmarkdown#257]
Here is a simple example that I can reproduce on my laptop.

_test.rmd_ contains Chinese and English words,and _testen.rmd_ only contains English words.

_test.rmd_ save as UTF-8 without BOM

<pre>

---
title: 词汇表
layout: default

---
test

# 词汇表 
这是一个测试。

```{r, eval = FALSE}
a="这是test。"
b
c
```
</pre>

<p></p>

_testen.rmd_

<pre>

---
title: voc
layout: default

---
test

# voc
This is a test.

```{r, eval = FALSE}
a="This is a test."
b
c
```
</pre>

<p></p>

The R script to run:

<pre>
> rmarkdown::render(input = "test.rmd",encoding = "UTF-8")
Error in nchar(front_matter) : invalid multibyte string 1
> rmarkdown::render(input = "testen.rmd",encoding = "UTF-8")

processing file: testen.rmd
  |................................                                 |  50%
  ordinary text without R code

  |.................................................................| 100%
label: unnamed-chunk-1 (with options) 
List of 1
 $ eval: logi FALSE

output file: testen.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" testen.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash-implicit_figures --output testen.html --smart --email-obfuscation none --self-contained --standalone --section-divs --template "C:\Users\outwen\R\win-library\3.1\rmarkdown\rmd\h\default.html" --variable "theme:bootstrap" --include-in-header "C:\Users\outwen\AppData\Local\Temp\RtmpgzFSrI\rmarkdown-str124073aa465c.html" --mathjax --variable "mathjax-url:https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" --no-highlight --variable highlightjs=testen_files/highlight 

Output created: testen.html
</pre>

```
> sessionInfo()
R version 3.1.1 (2014-07-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bookdown_0.1

loaded via a namespace (and not attached):
[1] digest_0.6.4    evaluate_0.5.5  formatR_1.0     htmltools_0.2.6
[5] knitr_1.6.12    rmarkdown_0.3.3 stringr_0.6.2   tools_3.1.1    
[9] yaml_2.1.13  
```
